spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DataFlowServerConfiguration.java

```diff
@@ -161,10 +175,97 @@ public void configurePathMatch(PathMatchConfigurer configurer) {
 	}
 
 	@Bean
-	@ConditionalOnMissingBean
+	@ConditionalOnBean(Server.class)
+	public TaskExplorer taskExplorer(DataSource dataSource, Server server) {
+		return new SimpleTaskExplorer(new TaskExecutionDaoFactoryBean(dataSource));
+	}
+
+	@Bean
+	@ConditionalOnMissingBean(Server.class)
 	public TaskExplorer taskExplorer(DataSource dataSource) {
-		JdbcTaskExplorerFactoryBean factoryBean = new JdbcTaskExplorerFactoryBean(dataSource);
-		return factoryBean.getObject();
+		return new SimpleTaskExplorer(new TaskExecutionDaoFactoryBean(dataSource));
+	}
+
+	@Bean
+	public TaskJobRepository taskJobExecutionRepository(JobService service, TaskExplorer taskExplorer){
+		return new TaskJobRepository(service, taskExplorer);
+	}
+
+	@Bean
+	public SimpleJobServiceFactoryBean simpleJobServiceFactoryBean(DataSource dataSource,
+			JobRepositoryFactoryBean repositoryFactoryBean) {
+		SimpleJobServiceFactoryBean factoryBean = new SimpleJobServiceFactoryBean();
+		factoryBean.setDataSource(dataSource);
+		try {
+			factoryBean.setJobRepository(repositoryFactoryBean.getObject());
+			factoryBean.setJobLocator(new MapJobRegistry());
+			factoryBean.setJobLauncher(new SimpleJobLauncher());
+			factoryBean.setDataSource(dataSource);
+		}
+		catch (Exception e) {
+			throw new IllegalStateException(e);
+		}
+		return factoryBean;
+	}
+
+	@Bean
+	@ConditionalOnBean(Server.class)
+	public JobRepositoryFactoryBean jobRepositoryFactoryBeanForServer(DataSource dataSource, Server server, DataSourceTransactionManager dataSourceTransactionManager){
```

Comment:
I understand now this is the h2 server. It feels very weird to me that some beans depend on a concrete datasource implementation type to work / even be created.